src/app/home/page.tsx

```diff
@@ -0,0 +1,79 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import { keepPreviousData, useQuery } from '@tanstack/react-query';
+import { useFilter } from '../../hooks/useFilter';
+import { Pagination } from './components/Pagination/Pagination';
+import { SearchForm } from './components/SearchForm/SearchForm';
+import { Loader } from '../../components/loader/Loader';
+import { SelectedFlyout } from './components/SelectedFlyout/SelectedFlyout';
+import { fetchCharacters } from '../../api/api';
+import { CharactersList } from './components/CharactersList/CharactersList';
+
+export type Character = {
+  id: number;
+  name: string;
+  status: string;
+  species: string;
+  gender: string;
+  location: {
+    name: string;
+  };
+  image: string;
+};
+
+export type ServerData = {
+  info: Record<string, unknown>;
+  results: Character[];
+};
+
+export default function HomePage() {
+  const [query, setQuery] = useState('');
+  const [totalPages, setTotalPages] = useState(0);
+  const [currentPage, setCurrentPage] = useState(1);
+  const { data, isLoading } = useQuery({
+    queryKey: ['characters', currentPage],
+    queryFn: () => fetchCharacters(currentPage),
+    placeholderData: keepPreviousData,
+  });
+  const filteredData = useFilter(data, query);
+
+  useEffect(() => {
+    if (data?.info?.pages) setTotalPages(data.info.pages);
+  }, [data]);
+
+  const handleQuery = (query: string): void => {
+    setQuery(query);
+    setCurrentPage(1);
+  };
+
+  const handlePageChange = (page: number) => {
+    if (page >= 1 && page <= totalPages && page !== currentPage) {
+      setCurrentPage(page);
```

Comment:
Currently the url isn't updated by page param
Take a look on this approach to update  url by next/navigation 
```
const router = useRouter();
const searchParams = useSearchParams();
const pathname = usePathname();

const updateUrlByDetails = (id: number) => {
const params = new URLSearchParams(searchParams.toString());
params.set('details', id.toString());
router.push(`${pathname}?${params.toString()}`);
};

```